src/pmdas/rocestat/pmdarocestat.python

```diff
@@ -0,0 +1,646 @@
+            dev = RoceDev()
+            for k,v in dict_counters.items():
+                setattr(dev, k, v)
+
+            self.mlnx_counters[device] = dev
+
+    def refresh_nic_stats(self):
+        try:
+            collector = NICSTATS()
+            nic_stats = collector.collect_stats()
+        except:
+            self.log("failed to fetch nic stats")
+
+        cached_nic_stats = {}
+        if bool(self.nic_stats):
+            for dev_lane, subdict in self.nic_stats.items():
+                if not dev_lane.endswith('delta'):
+                    prev_values = {}
+                    for stat_tuple in NicStats._fields_:
+                        prev_values[stat_tuple[0]] = getattr(subdict, stat_tuple[0])
+
+                    cached_nic_stats[dev_lane] = prev_values
+        else:
+            self.log("No per lane stats in the cache")
+
+        bw = lambda x,y: float("%0.2f" % ((float(y)-float(x))*8.0e-9/deltat))
+        per_sec = lambda x,y: (int(x)-int(y))/deltat if deltat > 0 else (int(x)-int(y))
+
+        if bool(cached_nic_stats):
+            for key in nic_stats.keys():
+                nic_stats[key]['tx_bytes_bw'] = bw(cached_nic_stats[key]['tx_bytes'], nic_stats[key]['tx_bytes'])
+                nic_stats[key]['rx_bytes_bw'] = bw(cached_nic_stats[key]['rx_bytes'], nic_stats[key]['rx_bytes'])
+                nic_stats[key]['rx_pause_delta'] = per_sec(nic_stats[key]['rx_pause'], cached_nic_stats[key]['rx_pause'])
+        else:
+            for key in nic_stats.keys():
+                nic_stats[key]['tx_bytes_bw'] = 0
+                nic_stats[key]['rx_bytes_bw'] = 0
+                nic_stats[key]['rx_pause_delta'] = 0
+
+        for nic_lane, dict_stats in nic_stats.items():
+            nic = NicStats()
+            for k,v in dict_stats.items():
+                setattr(nic, k, v)
+
+            self.nic_stats[nic_lane] = nic
+
+    def refresh(self):
+        try:
+            self.refresh_mlnx_counters()
+            try:
+                self.refresh_nic_stats()
+            except:
+                self.log("Nic stats refresh failed")
+        except:
+            self.log("mlnx counters refresh failed")
+
+    def roce_stat_fetch(self):
+        ''' Called once per "fetch" PDU '''
+        self.refresh()
+        self.replace_indom(self.mlnx_indom, self.mlnx_counters)
+        self.replace_indom(self.nic_indom, self.nic_stats)
+
+    def roce_fetch_callback(self, cluster, item, inst):
+        """
+        Main fetch callback
+        """
+
+        if CLUSTER_HW <= cluster <= CLUSTER_HW_LINK:
+            return self.rocestat_fetch_mlnx_counters(item, inst)
+
+        if cluster == CLUSTER_NIC:
+            return self.rocestat_fetch_nic_stats(item, inst)
+
+        return [c_api.PM_ERR_PMID, 0]
+
+
+    def roce_store_callback(self, cluster, item, inst, val):
+        """
+        Store callback, executed when a request to write to a metric happens
+        """
+        self.log("storing values\n")
```

Comment:
Looks like we can safely remove this store function (and registering it below) in `__init__` function.